src/thinblock.h

```diff
@@ -10,18 +10,21 @@
 #include "primitives/block.h"
 #include "bloom.h"
 #include "stat.h"
+#include "sync.h"
 #include "consensus/validation.h"
 #include "protocol.h"
 #include <vector>
 
+using namespace std;
```

Comment:
please do not "using namespace..." in header files, because it clutters the namespace for everyone who includes your header:
http://stackoverflow.com/questions/5849457/using-namespace-in-c-headers